Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,24 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
+  glue = options[:glue] || "\n[...snip...]\n"
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
+  scale = 0.99 # just a tad small to allow for the glue
+  while true
```

Comment:
If you convert to 7-bit ASCII and back (lossily) you are guaranteed to be correct, both in size and encoding validity. But I think that will lose useful diagnostic information. There are enough non-US programs and users in Homebrew that non-US characters – even just accented vowels, or names of font files – might well make it in to build script output. No non-US characters will survive the round-trip, which is a bummer.

And if "weird" characters do make it in to a build script's execution, that might cause problems, and being able to see that clearly might be helpful in diagnosing them.

Also because of the Ruby encoding issues, I'm not entirely sure about the behavior of round-tripping strings through 7-bit ASCII under Ruby 1.8 vs 1.9 vs 2.0.